ecosystem/gstreamer_plugin/gst_mtl_common.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include "gst_mtl_common.h"
+
+gboolean gst_mtl_common_parse_input_finfo(const GstVideoFormatInfo* finfo,
+                                          enum st_frame_fmt* fmt) {
+  if (finfo->format == GST_VIDEO_FORMAT_v210) {
+    *fmt = ST_FRAME_FMT_V210;
+  } else if (finfo->format == GST_VIDEO_FORMAT_I420_10LE) {
+    *fmt = ST_FRAME_FMT_YUV422PLANAR10LE;
+  } else {
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+gboolean gst_mtl_common_parse_fps_code(gint fps_code, enum st_fps* fps) {
+  if (!fps) {
+    GST_ERROR("Invalid fps pointer");
+    return FALSE;
+  }
+
+  switch (fps_code) {
+    case GST_MTL_SUPPORTED_FPS_120:
+      *fps = ST_FPS_P120;
+      break;
+    case GST_MTL_SUPPORTED_FPS_119_88:
+      *fps = ST_FPS_P119_88;
+      break;
+    case GST_MTL_SUPPORTED_FPS_100:
+      *fps = ST_FPS_P100;
+      break;
+    case GST_MTL_SUPPORTED_FPS_60:
+      *fps = ST_FPS_P60;
+      break;
+    case GST_MTL_SUPPORTED_FPS_59_94:
+      *fps = ST_FPS_P59_94;
+      break;
+    case GST_MTL_SUPPORTED_FPS_50:
+      *fps = ST_FPS_P50;
+      break;
+    case GST_MTL_SUPPORTED_FPS_30:
+      *fps = ST_FPS_P30;
+      break;
+    case GST_MTL_SUPPORTED_FPS_29_97:
+      *fps = ST_FPS_P29_97;
+      break;
+    case GST_MTL_SUPPORTED_FPS_25:
+      *fps = ST_FPS_P25;
+      break;
+    case GST_MTL_SUPPORTED_FPS_24:
+      *fps = ST_FPS_P24;
+      break;
+    case GST_MTL_SUPPORTED_FPS_23_98:
+      *fps = ST_FPS_P23_98;
+      break;
+    default:
+      return FALSE;
+  }
+
+  return TRUE;
+}
+
+gboolean gst_mtl_common_parse_fps(GstVideoInfo* info, enum st_fps* fps) {
+  gint fps_div;
+  if (info->fps_n <= 0 || info->fps_d <= 0) {
+    return FALSE;
+  }
+
+  fps_div = info->fps_n / info->fps_d;
+
+  switch (fps_div) {
+    case 24:
+      *fps = ST_FPS_P24;
+      break;
+    case 25:
+      *fps = ST_FPS_P25;
+      break;
+    case 30:
+      *fps = ST_FPS_P30;
+      break;
+    case 50:
+      *fps = ST_FPS_P50;
+      break;
+    case 60:
+      *fps = ST_FPS_P60;
+      break;
+    case 120:
+      *fps = ST_FPS_P120;
+      break;
+    default:
+      return FALSE;
+  }
+
+  return TRUE;
+}
+
+/* includes all formats supported by the library for future support */
+gboolean gst_mtl_common_parse_pixel_format(const char* format, enum st_frame_fmt* fmt) {
+  if (!fmt || !format) {
+    GST_ERROR("%s, invalid input\n", __func__);
+    return FALSE;
+  }
+
+  if (strcmp(format, "YUV422PLANAR10LE") == 0) {
```

Comment:
We could optimize this "if" structure, so we check first "format" length and then compare the strings accordingly to this length.
It would shorten the computational complexity of the procedure.

Reply:
Is it worth it? 
This is 0,1 s of initialization time when we have more than 10 seconds in the initialization of the backend.